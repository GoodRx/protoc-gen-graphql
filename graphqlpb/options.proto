syntax = "proto3";

package graphql;

option go_package = "github.com/martinxsliu/protoc-gen-graphql/graphqlpb";

import "google/protobuf/descriptor.proto";

extend google.protobuf.MethodOptions {
  string operation = 82731;
}
