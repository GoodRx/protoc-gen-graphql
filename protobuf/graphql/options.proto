syntax = "proto2";

package graphql;

option go_package = "github.com/martinxsliu/protoc-gen-graphql/protobuf/graphql";

import "google/protobuf/descriptor.proto";

extend google.protobuf.FileOptions {
  optional FileOptions file = 82731;
}

extend google.protobuf.MessageOptions {
  optional MessageOptions message = 82731;
}

extend google.protobuf.FieldOptions {
  optional FieldOptions field = 82731;
}

extend google.protobuf.EnumOptions {
  optional EnumOptions enum = 82731;
}

extend google.protobuf.EnumValueOptions {
  optional EnumValueOptions enum_value = 82731;
}

extend google.protobuf.ServiceOptions {
  optional ServiceOptions service = 82731;
}

extend google.protobuf.MethodOptions {
  optional MethodOptions method = 82731;
}

message FileOptions {
  // Name used to prefix all types with. If unset, the default behaviour is
  // to use the upper camel case of the Protobuf package name.
  optional string namespace = 1;
}

message MessageOptions {
  // Name of the generated GraphQL type.
  optional string type = 1;
}

message FieldOptions {
  // Name of the field in the generated GraphQL object and input types.
  optional string field = 1;

  // Name of GraphQL type to use for this field, including modifiers.
  optional string type = 2;

  // Skip this field in the generated GraphQL type.
  optional bool skip = 3;

  // GraphQL directive to generate for the field. Do not include the @ sign,
  // do include any arguments within parentheses.
  repeated string directive = 4;

  // Mark the field as a foreign key to another Protobuf message. The value
  // should be the name of the Protobuf type, following the same type name
  // resolution rules, like when using types imported from other packages.
  optional string foreign_key = 5;
}

message EnumOptions {
  // Name of the generated GraphQL type.
  optional string type = 1;
}

message EnumValueOptions {
  // Name of the enum value in the generated GraphQL enum type.
  optional string value = 1;
}

message ServiceOptions {
  // A variable-like name to reference the service with, in lieu of the
  // name generated from the service's package and service name. The name
  // should be unique across all gRPC services.
  // Note: This will not be the name of the generated GraphQL type.
  optional string reference_name = 1;
}

message MethodOptions {
  // Name of the field in the GraphQL type generated for the service.
  optional string field = 1;

  // Type of GraphQL operation. Valid values are: "query", "mutation",
  // or "subscription". Defaults to "query".
  optional string operation = 2;

  optional string dataloader = 3;
}
