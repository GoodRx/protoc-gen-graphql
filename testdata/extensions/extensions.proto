syntax = "proto3";

package protoc_gen_graphql.test.extensions;

import "graphql/options.proto";

service Foo {
  option (graphql.service_options).reference_name = "foo";

  rpc QueryMethod(Request) returns (Response) {
    option (graphql.method_options).operation = "query";
  }
  rpc MutationMethod(Request) returns (Response) {
    option (graphql.method_options).operation = "mutation";
  }
  rpc SubscriptionMethod(Request) returns (Response) {
    option (graphql.method_options).operation = "subscription";
  }
  rpc UnknownMethod(Request) returns (Response) {
    option (graphql.method_options).operation = "unknown";
  }
}

message Request {
  string id = 1;
}

message Response {
  string name = 1;
}
