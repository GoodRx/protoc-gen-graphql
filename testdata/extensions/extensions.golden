# DO NOT EDIT! Generated by protoc-gen-graphql.

extend type Query {
  foo: ProtocGenGraphqlTestExtensions_Foo_Query
}

type ProtocGenGraphqlTestExtensions_Foo_Query {
  query_method(input: ProtocGenGraphqlTestExtensions_RequestInput!): ProtocGenGraphqlTestExtensions_Response!
  unknown_method(input: ProtocGenGraphqlTestExtensions_RequestInput!): ProtocGenGraphqlTestExtensions_Response!
}

extend type Mutation {
  foo: ProtocGenGraphqlTestExtensions_Foo_Mutation
}

type ProtocGenGraphqlTestExtensions_Foo_Mutation {
  mutation_method(input: ProtocGenGraphqlTestExtensions_RequestInput!): ProtocGenGraphqlTestExtensions_Response!
}

extend type Subscription {
  foo: ProtocGenGraphqlTestExtensions_Foo_Subscription
}

type ProtocGenGraphqlTestExtensions_Foo_Subscription {
  subscription_method(input: ProtocGenGraphqlTestExtensions_RequestInput!): ProtocGenGraphqlTestExtensions_Response!
}

type ProtocGenGraphqlTestExtensions_Request {
  id: String!
}

input ProtocGenGraphqlTestExtensions_RequestInput {
  id: String!
}

type ProtocGenGraphqlTestExtensions_Response {
  name: String!
}
